src/structures/Guild.js

```diff
@@ -68,8 +68,8 @@ class Guild {
        */
       this.id = data.id;
     } else {
-      this.available = true;
       this.setup(data);
+      if (!data.channels) this.available = false;
```

Comment:
To clarify: `this.setup` sets the `available` property to `true` here, see [line 167](https://github.com/SpaceEEC/discord.js/blob/63a6f49d6e3d9c27ec99518f6c836d82c143ed39/src/structures/Guild.js#L167).
If there are no channels it will change it to false after the setup.